photon-ml/src/main/scala/com/linkedin/photon/ml/avro/model/ModelProcessingUtils.scala

```diff
@@ -109,7 +118,8 @@ object ModelProcessingUtils {
         val Array(featureShardId) = IOUtils.readStringsFromHDFS(idInfoPath, configuration).toArray
 
         // Load the coefficients
-        val featureNameAndTermToIndexMap = featureShardIdToIndexMapLoader(featureShardId).indexMapForDriver()
+        val indexMapLoaders = featureShardIdToIndexMapLoader.flatMap(_.get(featureShardId))
+        val featureNameAndTermToIndexMap = indexMapLoaders.map(_.indexMapForDriver())
```

Comment:
Yeah - we have type inference, and Scala borrows from Python, which hides types as much as it can, so I'm guessing the philosophy is more to make life easier for devs by figuring out the types for them as much as possible. In my opinion, avoiding type annotations allows to better focus on the logic of the code.